pkg/volume/csi/csi_client.go

```diff
@@ -77,29 +77,52 @@ func newCsiDriverClient(network, addr string) *csiDriverClient {
 	return &csiDriverClient{network: network, addr: addr}
 }
 
-// assertConnection ensures a valid connection has been established
-// if not, it creates a new connection and associated clients
-func (c *csiDriverClient) assertConnection() error {
-	if c.conn == nil {
-		conn, err := grpc.Dial(
-			c.addr,
-			grpc.WithInsecure(),
-			grpc.WithDialer(func(target string, timeout time.Duration) (net.Conn, error) {
-				return net.Dial(c.network, target)
-			}),
-		)
-		if err != nil {
-			return err
-		}
-		c.conn = conn
-		c.idClient = csipb.NewIdentityClient(conn)
-		c.nodeClient = csipb.NewNodeClient(conn)
-		c.ctrlClient = csipb.NewControllerClient(conn)
+func (c *csiDriverClient) IdentityRequest() (*grpc.ClientConn, csipb.IdentityClient, error) {
+	conn, err := grpc.Dial(
+		c.addr,
+		grpc.WithInsecure(),
+		grpc.WithDialer(func(target string, timeout time.Duration) (net.Conn, error) {
+			return net.Dial(c.network, target)
+		}),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.idClient = csipb.NewIdentityClient(conn)
+
+	return conn, c.idClient, nil
+}
+
+func (c *csiDriverClient) NodeRequest() (*grpc.ClientConn, csipb.NodeClient, error) {
+	conn, err := grpc.Dial(
+		c.addr,
+		grpc.WithInsecure(),
+		grpc.WithDialer(func(target string, timeout time.Duration) (net.Conn, error) {
+			return net.Dial(c.network, target)
+		}),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.nodeClient = csipb.NewNodeClient(conn)
+
+	return conn, c.nodeClient, nil
+}
 
-		// set supported version
+func (c *csiDriverClient) ControllerRequest() (*grpc.ClientConn, csipb.ControllerClient, error) {
```

Comment:
I would call method `ControllerClient`, instead of ControllerRequest, since it's used to create the client.

Reply:
will do

Reply:
How about `NewControllerClient()`?